modules/http4s/src/main/scala/com.snowplowanalytics.iglu/client/resolver/registries/Http4sRegistryLookup.scala

```diff
@@ -94,5 +99,44 @@ object Http4sRegistryLookup {
       .fromString(s"${cxn.uri.toString.stripSuffix("/")}/schemas/$vendor/$name/jsonschema/$model")
       .leftMap(e => RegistryError.ClientFailure(e.message))
 
+  def runRequest[F[_]: Sync, A: EntityDecoder[F, *]](
+    client: HttpClient[F],
+    req: Request[F]
+  ): F[Either[RegistryError, A]] = {
+    val responseResult = client.run(req).use[F, Either[RegistryError, A]] {
+      case Status.Successful(response) =>
+        response
+          .as[A]
+          .map(_.asRight[RegistryError])
+          .handleError {
```

Comment:
I know this is exactly what I just said to do..... but I'm looking at it again and wondering if it was the right thing 🤦‍♂️.  Because you already handle the error later on at line 134.  And also it's inconsistent, because there are other places where you don't catch errors that might get raised (e.g. `response.bodyText.compile.string`).  I guess you've added an informative error message, so maybe it is better.

Oh also..... I _think_ that methods like `handleError` are guaranteed to be called with non-fatal exceptions only.  So it shouldn't be necessary to pattern match on `NonFatal`.  However, I never found good documentation on this - I think I just read it somewhere in a gitter channel.


I don't suggest changing anything though.  This is all good.

Reply:
Yeah, I think there's still a slight difference that would help us to catch trimmed payloads.

On line 134 I catch very IO-ish _client_ exceptions, i.e. network is unavailable, connection pool issue etc. Whereas this `handleError` is specifically about things like decoding failure due trimming or dropped connection and we'll get different error messages.

Although that's post-rationalisation - I also haven't realised the exceptions is caught.